python-fs-providers/box-com_box-com/fs-provider.py

```diff
@@ -7,6 +7,76 @@
 from utils import get_full_path, get_rel_path, get_normalized_path
 
 
+# The DefaultNetwork of box.com logs all network queries and responses, INCLUDING CONTENT
+# This is crazy but it is how it is, and it of course lead to incredible disk usage
+# and various failures because these insane log lines can find their way to a LogTail in DSS
+#
+# The DefaultNetwork has provision to avoid logging when the response is a stream
+# (see response_as_stream in DefaultNetworkResponse), but boxsdk.object.file always uses
+# the .content instead of the .raw, so it always goes through logging
+#
+# So we override the DefaultNetwork's log function to avoid logging big things
+#
+# Note: another option would be to try to use response_as_stream instead, but it would
+# require rewrapping boxsdk.object.file, so looks more dangerous
+MAX_LOGGED_RESPONSE_SIZE = 2048
+
+from boxsdk.network.default_network import DefaultNetwork, DefaultNetworkResponse
+from boxsdk.session.session import AuthorizedSession
+from six import text_type
+from pprint import pformat
+from boxsdk.util.log import sanitize_dictionary
+
+class LessVerboseLoggingNetwork(DefaultNetwork):
+    @property
+    def network_response_constructor(self):
+        return LessVerboseLoggingNetworkResponse
+
+class LessVerboseLoggingNetworkResponse(DefaultNetworkResponse):
+    def log(self, can_safely_log_content=False):
+        if self._did_log:
+            return
+        self._did_log = True
+        content_length = self.headers.get('Content-Length', None)
+        content = self.STREAM_CONTENT_NOT_LOGGED
+
+        if can_safely_log_content:
+            if content_length is None:
+                content_length = text_type(len(self.content))
+
+            if len(self.content) > MAX_LOGGED_RESPONSE_SIZE:
+                #print("DISABLING LOGGING, content is too large: %s" % content_length)
```

Comment:
unused comment

Reply:
It's interesting, I actually disagree with the precept that code should be free of commented-out statements, especially commented-out logging statements, as it allows readding them easily when needed :) (I won't fight about that of course :) )

Reply:
Oh, thanks for letting know, I decided not to keep the PR open with this comment for too long since the issue was pretty important and the release should've been made ASAP as I understood. 

Interesting to know your opinion on this since I was used to the opposite.